File: crop_logo.py
import sys
import subprocess

# Try to install Pillow if it's not installed
try:
    from PIL import Image
except ImportError:
    print("Pillow not installed. Installing Pillow now...")
    subprocess.check_call([sys.executable, "-m", "pip", "install", "Pillow"])
    from PIL import Image

try:
    img = Image.open('assets/icons/logo.png')
    img = img.convert('RGBA')
    
    # Get bounding box of non-white pixels
    # First, convert to grayscale to find white pixels easily
    gray = img.convert('L')
    # Anything below 250 is considered non-white
    bw = gray.point(lambda x: 0 if x > 250 else 255, '1')
    bbox = bw.getbbox()
    
    if bbox:
        img = img.crop(bbox)
        img.save('assets/icons/logo.png')
        print('Logo cropped successfully! The white outline has been removed.')
    else:
        print('Could not find non-white pixels. Image might be completely white or already cropped.')
except Exception as e:
    print(f"An error occurred: {e}")

File: crop_logo.dart
import 'dart:io';
import 'package:image/image.dart' as img;

void main() {
  final file = File('assets/icons/logo.png');
  if (!file.existsSync()) {
    print('assets/icons/logo.png not found');
    return;
  }
  
  final bytes = file.readAsBytesSync();
  var image = img.decodeImage(bytes);
  
  if (image == null) {
    print('Failed to decode image');
    return;
  }
  
  // Ensure the image has an alpha channel
  if (image.numChannels != 4) {
    // Note: in image v4, we can just use the image directly, but setting alpha on a 3-channel image might not stick.
    // We can convert it to 4 channels.
    image = image.convert(numChannels: 4);
  }
  
  // Find bounding box to crop empty white/transparent space
  int minX = image.width;
  int minY = image.height;
  int maxX = 0;
  int maxY = 0;
  
  for (final pixel in image) {
    final r = pixel.r;
    final g = pixel.g;
    final b = pixel.b;
    
    // If it's a white pixel, make it transparent
    if (r >= 240 && g >= 240 && b >= 240) {
      pixel.a = 0;
    } else if (pixel.a > 0) {
      // It's a non-white, non-transparent pixel
      if (pixel.x < minX) minX = pixel.x.toInt();
      if (pixel.y < minY) minY = pixel.y.toInt();
      if (pixel.x > maxX) maxX = pixel.x.toInt();
      if (pixel.y > maxY) maxY = pixel.y.toInt();
    }
  }
  
  if (minX <= maxX && minY <= maxY) {
    // Crop the image to the bounding box
    final cropped = img.copyCrop(image, x: minX, y: minY, width: maxX - minX + 1, height: maxY - minY + 1);
    file.writeAsBytesSync(img.encodePng(cropped));
    print('Logo successfully cropped and white background removed!');
  } else {
    // If we didn't find any non-white pixels, just save the transparent image
    file.writeAsBytesSync(img.encodePng(image));
    print('Made white pixels transparent, but did not crop (image is mostly white).');
  }
}
